staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -159,6 +160,49 @@ func (i *indexedWatchers) terminateAll(objectType reflect.Type, done func(*cache
 	}
 }
 
+// As we don't need a high precision here, we keep all watchers timeout within a
+// second in a bucket, and pop up them once at the timeout. To be more specific,
+// if you set fire time at X, you can get the bookmark within (X-1,X+1) period.
+// This is NOT thread-safe.
+type timeBucketWatchers struct {
```

Comment:
Since `timeBucketWatchers.addWatcher` directly calls `w.nextBookmarkTime`, this implementation is tightly coupled with the notion of bookmarks. Would it be clearer to put "bookmark" in the name? e.g. `watcherBookmarkTimeBuckets`?

Reply:
take you advices